Documentation/ExtensionArchitecture/HowTo/Configuration.rst

```diff
@@ -0,0 +1,100 @@
+.. include:: /Includes.rst.txt
+.. index:: Extension; Configuration
+.. _extension_configuration:
+
+=============
+Configuration
+=============
+
+There are several possibilities to make your extension configurable. From the
+various options described here each differs in:
+
+*   the scope to what the configuration applies (extension, pages,
+    plugin)
+*   the access level required to make the change (editor, admin)
+
+.. index::
+   TypoScript
+   Configuration; TypoScript
+
+TypoScript and constants
+========================
+
+You can define configuration options using TypoScript.
+These options can be changed via TypoScript constants and setup in the backend.
+The changes apply to the current page and all subpages.
+
+.. seealso::
+
+   * :ref:`extbase_typoscript_configuration`
+
+   * :ref:`t3tsref:typoscript-syntax-what-are-constants`
+
+
+.. index::
+   Configuration; Extension configuration
+   Files; ext_conf_template.txt
+
+Extension configuration
+=======================
+
+Extension configuration is defined in the file :file:`ext_conf_template.txt`
+using TypoScript constant syntax.
+
+The configuration options you define in this file can be changed in the
+backend :guilabel:`Admin Tools > Settings > Extension Configuration`.
+
+Use this file for general options that should be globally applied to the extension.
+
+.. seealso::
+
+   * :ref:`extension-options`
+
+
+.. index:: FlexForms
+
+FlexForms
+=========
+
+FlexForms can be configured in the backend by editors. With FlexForms you can
+configure each plugin or content element individually.
```

Comment:
```suggestion
configure each plugin or content element individually without adding extra fields to the :sql:`tt_content` table.
```
We should make clear the advantages and disadvantages: Using FlexForm for settings (`settings.`) FlexForm is okay. For using extra fields, maybe enhancing the tt_content table or reusing other fields are more appropriate.

Reply:
This chapter is about Extension configuration so I don't think missuse of FlexForms to store data is really in the scope here